packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -437,12 +440,13 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
         ? composeRenderFunction(props.onRenderRow, onRenderDefaultRow)
         : onRenderDefaultRow;
 
+      const numOfGroupHeadersBeforeItem: number = getNumGroupHeaders(groups, index);
```

Comment:
We should avoid any code which enumerates any number of `groups` for each item we're about to render.
It looks like we need to do two things:
1. Define a mapping of `group.startIndex` to the index of the group.
2. Pass `group` to `onRenderCell`, so we can get `group.startIndex` and do the lookup.

Reply:
The `useGroupedDetailsListIndexMap` function does something similar and returns a mapping that this can use to lookup `numGroupHeaders` based on the `group`'s unique `key`